soc/nordic/nrf54h/soc.c

```diff
@@ -172,11 +179,24 @@ void soc_late_init_hook(void)
 	 */
 	uint8_t *msg = NULL;
 	size_t msg_size = 0;
-
-	void *radiocore_address =
-		(void *)(DT_REG_ADDR(DT_GPARENT(DT_NODELABEL_CPURAD_SLOT0_PARTITION)) +
-			 DT_REG_ADDR(DT_NODELABEL_CPURAD_SLOT0_PARTITION) +
-			 CONFIG_ROM_START_OFFSET);
+	void *radiocore_address = NULL;
+
+	if (PARTITION_IS_RUNNING_APP_PARTITION(slot0_partition)) {
+		radiocore_address =
+			(void *)(DT_REG_ADDR(DT_GPARENT(DT_NODELABEL(cpurad_slot0_partition))) +
+				 DT_REG_ADDR(DT_NODELABEL(cpurad_slot0_partition)) +
+				 CONFIG_ROM_START_OFFSET);
+#if DT_NODE_EXISTS(DT_NODELABEL(cpurad_slot1_partition))
+	} else if (PARTITION_IS_RUNNING_APP_PARTITION(slot1_partition)) {
```

Comment:
This is missing the case where Application core firmware runs from `boot_partition`, which is when it isn't started by MCUboot. We expect to start Radiocore from `cpurad_slot0_partition` in general.

How about changing it like this?
```c
if (DT_NODE_EXISTS(DT_NODELABEL(cpurad_slot1_partition)) &&
    PARTITION_IS_RUNNING_APP_PARTITION(slot1_partition)) {
	/* cpurad_slot1_partition */
} else {
	/* cpurad_slot0_partition */
}
```